--- a/src/lib.rs	2025-09-23 13:41:09.327582205 +0200
+++ b/src/lib.rs	2025-09-23 13:41:23.043271856 +0200
@@ -241,7 +241,12 @@
 #[doc(hidden)]
 mod private;
 
-include!(concat!(env!("OUT_DIR"), "/private.rs"));
+#[doc(hidden)]
+pub mod __private_MESON {
+    #[doc(hidden)]
+    pub use crate::private::*;
+}
+use serde_core::__private_MESON as serde_core_private;
 
 // Re-export #[derive(Serialize, Deserialize)].
 //
